Iot/bf/pi node and gateway

bevanyeah opened the pull request:
Image (jpg) splitter and recompiler

Place a new file (name as a number.jpg) into the Node root directory.  It will negotiate between the receiver with a header of the new file, and the receiver will request the components of the file 1 at a time.  it will automatically retry missed packets.

When the file components are complete, it will reconstruct the jpg file and store in local storage.

bevanyeah committed:
Initial framework for underlying Node and Receiver Code

bevanyeah committed:
Bytes being correctly sent and recompiled at the other end

bevanyeah committed:
Resolved jpg recompiling issues

scorpio-hank reviewed (approved):
Looks awesome, will try adapy my code in ASAP

scorpio-hank committed:
Update battery code in PiNode

This code integrates the original Arduino code (direct battery monitoring) with the INA219 chip to calculate battery level

bevanyeah committed:
Recommit of all changes following failed commit

bevanyeah committed:
Merge branch 'iot/bf/pi_node_and_gateway' of https://github.com/DataBytes-Organisation/Project-Echo-IoT into iot/bf/pi_node_and_gateway
